src/server/array_func.c

```diff
@@ -422,6 +422,17 @@ chk_array_doneness(job *parent)
 		 */
 		svr_saveorpurge_finjobhist(parent);
 	} else {
+		/* Before we do a full save of parent, recalculate "JOB_ATR_array_indices_remaining" here*/
+		attribute *premain = &parent->ji_wattr[(int)JOB_ATR_array_indices_remaining];
+		if (premain->at_flags & ATR_VFLAG_MODCACHE) {
+			char *pnewstr = cvt_range(parent->ji_ajtrk, JOB_STATE_QUEUED);
+			if (pnewstr == NULL)
+				pnewstr = "-";
+			job_attr_def[JOB_ATR_array_indices_remaining].at_free(premain);
+			job_attr_def[JOB_ATR_array_indices_remaining].at_decode(premain, 0, 0, pnewstr);
+			/* also update value of attribute "array_state_count" */
```

Comment:
You could actually make use of the function set_attr_svr() which automatically takes care of free and decoding and setting the attribute. This way there is no need of the variable premain also.

Reply:
I tried using the set_attr_svr() but it is crashing at assert() at the beginning of set_str() function called from set_attr_svr(). This is because set_attr_svr() cannot handle the case when value is empty string.

Reply:
set_attr_svr() makes use of our attribute framework functions. So it is the underlying functions in this case set_attr() should handle it accordingly. If not you can make use of decode() function only.

Reply:
Ok then I will leave it as it is now

Reply:
I am fine with the code as it stands